src/foundry/client-esm/applications/ux/text-editor.d.mts

```diff
@@ -1,9 +1,530 @@
+   * // Returns: { uuid: "Actor.xyz", caption: "Example Caption", values: ["inline"] }
+   * ```
+   */
+  protected static _parseEmbedConfig(
+    raw: string,
+    options?: foundry.utils.ParseUUIDOptions,
+  ): TextEditor.DocumentHTMLEmbedConfig;
+
+  /**
+   * Replace a hyperlink-like string with an actual HTML &lt;a&gt; tag
+   * @param match   - The regular expression match
+   * @param options - Options provided to customize text enrichment
+   * @returns An HTML element for the document link
+   */
+  protected static _createHyperlink(
+    match: RegExpMatchArray,
+    options?: TextEditor.EnrichmentOptions,
+  ): Promise<HTMLAnchorElement>;
+
+  /**
+   * Replace an inline roll formula with a rollable &lt;a&gt; element or an eagerly evaluated roll result
+   * @param match    - The regular expression match array
+   * @param rollData - Provided roll data for use in roll evaluation
+   * @param options  - Options provided to customize text enrichment (default: `{}`)
+   * @returns The replaced match. Returns null if the contained command is not a valid roll expression.
+   * @internal
+   */
+  protected static _createInlineRoll(
+    match: RegExpMatchArray,
+    rollData: AnyObject,
+    options?: TextEditor.EnrichmentOptions,
+  ): Promise<HTMLAnchorElement | null>;
+
+  /**
+   * Activate listeners for the interior content of the editor frame.
+   */
+  static activateListeners(): void;
+
+  /**
+   * Handle left-mouse clicks on an inline roll, dispatching the formula or displaying the tooltip
+   * @param event - The initiating click event
+   */
+  protected static _onClickInlineRoll(event: MouseEvent): void;
+
+  /**
+   * Extract JSON data from a drag/drop event.
+   * @param event - The drag event which contains JSON data.
+   * @returns The extracted JSON data. The object will be empty if the DragEvent did not contain JSON-parseable data.
+   * @remarks `TokensLayer#_onDropActorData` returns a number - a notification ID - the  if the user lacks permissions
+   * @remarks `User#assignHotbarMacro` returns a promise to a user document
+   */
+  static getDragEventData(event: DragEvent): JSONValue;
+
+  /**
+   * Given a Drop event, returns a Content link if possible such as `"@Actor[ABC123]"`, else `null`
+   * @param eventData - The parsed object of data provided by the transfer event
+   * @param options   - Additional options to configure link creation.
+   */
+  static getContentLink(eventData: JSONValue, options?: TextEditor.GetContentLinkOptions): Promise<string | null>;
+
+  /**
+   * Upload an image to a document's asset path.
+   * @param uuid - The document's UUID.
+   * @param file - The image file to upload.
+   * @returns The path to the uploaded image.
+   * @internal
+   */
+  protected static _uploadImage(uuid: string, file: File): Promise<string | void>;
+
+  /**
+   * Retrieve the configured TextEditor implementation.
+   * @remarks TODO: Link up with CONFIG
```

Comment:
You can just do `type Implementation = typeof CONFIG.ux.TextEditor;` in the namespace. If this requires stubs in `CONFIG` go for it.